### DISCUSSION TITLE
Add info, that translation is outdated (enhanced fallback content)

### DISCUSSION THREAD

**Discussion by trueberryless** [withastro/starlight#2497] [Feature Requests]
### What version of `starlight` are you using?

0.28.3

### What is your idea?

Currently the translation only checks if the file exists in other locals and shows the fallback content + info that the page is not available in your local when it can't be found.

The idea is, that Starlight also shows a info (not so severe as a warning) when a translation is out of date, which is far more complicated (because then Starlight needs to calculated with dates based on the Git History and then the git checkout must be with far enough history and what if the maintainer doesnt even use git and so on...).

Like the warning in the picture but more something like "This content is outdated, read the English version if you want up-to-date information!" (The german text in the image translates to something like: "This content is not yet available in your language")

![image](https://github.com/user-attachments/assets/ae250148-da19-4c11-8595-e815c7d5d437)

There are a lot of issues with this feature, but I'll mention it nonetheless because the "Last updated" feature also exists.

### Why is this feature necessary?

I just would love the additional accessibility for the user but we have to trade of some computing power I think cause this feature would mean running additional build calculations which are not that environmentally-friendly...

### Participation

- [ ] I am willing to submit a pull request for this feature.

**Comment by HiDeoo** [comment]
> Starlight needs to calculated with dates based on the Git History and then the git checkout must be with far enough history and what if the maintainer doesnt even use git and so on...

One additional issue is that a user could update an outdated translation only to fix a typo in this translation. This updates should probably not mark the translation as updated. You can also have the same scenario in the default language version where a commit only fix a typo in this version and should not mark all updated translations as outdated.

At this point, you need some kind of constraints, rules or even framework to handle all possible scenarios, e.g. something like [Lunaria](https://lunaria.dev/). I feel like this would be a better place to have a component or some kind of Starlight override in its [Starlight plugin](https://lunaria.dev/integrations/starlight/) to handle all these scenarios with a defined set of rules.

**Reply by trueberryless** [reply]
Yeah, you are totally right. There are too many different scenarios which would blow up Starlight core functionalities on its own.

Even Lunaria has It's limitations and isn't perfect because there are just too many scenarios. The Continuous Integration - in the sense of GitHub PRs - would need a variety of labels to understand what locales to set to completed, outdated or still missing...

Making a plugin for enhancing this information is a really good idea, maybe I'll look into it sometimes... (but probably not 😂)
